game-core/src/main/java/games/strategy/engine/framework/startup/mc/ClientModel.java

```diff
@@ -268,18 +266,17 @@ private IServerStartupRemote getServerStartup() {
    * Resets stats and nulls out references, keeps chat alive.
    */
   public void cancel() {
-    if (messenger == null) {
-      return;
-    }
-    objectStreamFactory.setData(null);
-    messenger.shutDown();
-    chatPanel.setChat(null);
     gameSelectorModel.setGameData(gameDataOnStartup);
     gameSelectorModel.setCanSelect(true);
-    hostIsHeadlessBot = false;
-    gameSelectorModel.setIsHostHeadlessBot(false);
-    gameSelectorModel.setClientModelForHostBots(null);
-    messenger.removeErrorListener(this);
```

Comment:
Just curious if re-ordering this method to be invoked earlier was required for the fix or just because it made more sense to group it above?

Reply:
The code in cancel was previously not executed because messenger was null.
But the buttons should get re-enabled even if messenger is null, so I just moved the other code into an if check and execute the re-enabling code in any case.

Reply:
Sorry for the confusion (I added that comment with whitespace changes ignored and it probably could have been better placed on a different line).

I meant line 282 specifically (`messenger.removeErrorListener(this);`).  It was previously executed after the `gameSelectorModel.set*()` calls, and now it's executed before it.  I was just curious if the removal of the error listener was moved before the `set*()` calls because there was an unwanted notification being fired or something.  (Those the _best_ kinds of bugs. :smile:)

Reply:
Ah makes sense, but again: No.
The order changes are all more or less unintentional. (I did not actively check if the order was preserved, but didn't actively try to swap everything around either)